src/System.Net.Http.WinHttpHandler/src/System/Net/Http/WinHttpHandler.cs

```diff
@@ -1290,7 +1290,7 @@ private void ThrowOnInvalidHandle(SafeWinHttpHandle handle)
         
         private Task<bool> InternalSendRequestAsync(WinHttpRequestState state)
         {
-            state.TcsSendRequest = new TaskCompletionSource<bool>();
+            state.TcsSendRequest = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
```

Comment:
As stated in the original PR: what would happen if the certificate callback blocks? Do we still want to only document that behavior or should we consider removing the callback?